src/main/kotlin/org/jitsi/nlj/dtls/DtlsStack.kt

```diff
@@ -182,6 +182,22 @@ class DtlsStack(
         dtlsTransport?.send(data, off, len)
     }
 
+    private fun processIncomingProtocolData() {
+        var bytesReceived: Int
+        do {
+            synchronized(this) {
+                bytesReceived = dtlsTransport?.receive(dtlsAppDataBuf, 0, 1500, 1) ?: -1
+            }
+            if (bytesReceived > 0) {
+                // Copy again to copy out of dtlsAppDataBuf, which we re-use.
+                val bufCopy2 = BufferPool.getBuffer(bytesReceived).apply {
+                    System.arraycopy(dtlsAppDataBuf, 0, this, 0, bytesReceived)
+                }
+                incomingDataHandler?.dataReceived(bufCopy2, 0, bytesReceived)
```

Comment:
Is it okay that this will run in the thread calling `start()`? I guess practically it doesn't matter much since it's just passed on to the sctp stack

Reply:
I mean the thread that's calling `start()` is already a blocking thread so taking more time shouldn't be an issue.

I made sure not to hold my new synchronization lock when calling the SCTP stack (which does have internal synchronization).

Are there other issues you're concerned about?